source/appModules/outlook.py

```diff
@@ -427,49 +415,54 @@ def _get_name(self):
-			except COMError:
-				attachmentCount=0
-			# Translators: when an email has attachments
-			if attachmentCount>0: textList.append(_("has attachment"))
-			try:
-				importance=selection.importance
-			except COMError:
-				importance=1
-			importanceLabel=importanceLabels.get(importance)
-			if importanceLabel: textList.append(importanceLabel)
 			try:
 				messageClass=selection.messageClass
 			except COMError:
 				messageClass=None
-			if messageClass=="IPM.Schedule.Meeting.Request":
-				# Translators: the email is a meeting request
-				textList.append(_("meeting request"))
+			if self.appModule.outlookVersion<15:
+				# Translators: when an email is unread
+				if unread: textList.append(_("unread"))
+				if messageClass=="IPM.Schedule.Meeting.Request":
+					# Translators: the email is a meeting request
+					textList.append(_("meeting request"))
+			elif messageClass is not None:
+				# Replied or forwarded state for this message is available from the object's value.
+				# We must parse this value correctly, as it may contain redundant information, such as the message class and read value.
+				# We only expose the unread state, and message class for non-messages.
+				# The several states are localized and separated by a space.
+				# Example output: 'Meeting request Replied Read'
+				valueParts = self._get_value().split(" ")
+				valueCount = len(valueParts)
+				# The last valuePart indicates whether the message is read or unread.
+				# Do not expose the read state
+				lastPart = valueCount if unread else valueCount-1
+				# The first valuePart is the type of the selection, e.g. Message, Contact, meeting request.
+				# We can safely assume that the message class of a regular message or appointment is one word.
+				# For messages other than regular mail messages, the message class is relevant.
+				# Contact folders may contain both contacts and distribution lists,
+				firstPart = 1 if messageClass in silentMessageClasses else 0
+				textList.extend(valueParts[firstPart:lastPart])
 		childrenCacheRequest=UIAHandler.handler.baseCacheRequest.clone()
 		childrenCacheRequest.addProperty(UIAHandler.UIA_NamePropertyId)
 		childrenCacheRequest.addProperty(UIAHandler.UIA_TableItemColumnHeaderItemsPropertyId)
 		childrenCacheRequest.TreeScope=UIAHandler.TreeScope_Children
-		childrenCacheRequest.treeFilter=UIAHandler.handler.clientObject.createPropertyCondition(UIAHandler.UIA_ControlTypePropertyId,UIAHandler.UIA_TextControlTypeId)
 		cachedChildren=self.UIAElement.buildUpdatedCache(childrenCacheRequest).getCachedChildren()
 		for index in xrange(cachedChildren.length):
 			e=cachedChildren.getElement(index)
-			# #6219: Outlook 2016 started exposing the draft column as a text node.
-			# Users reportedly find this extremely annoying.
-			# Thus we filter it out.
-			if self.appModule.outlookVersion>=16 and e.cachedClassName=="DraftFlagField":
-				continue
+			isFlag = e.cachedClassName=="FlagField"

```

Comment:
Does this handle the "DraftFlagField" case?

Reply:
Good point, I forgot to mention this in the initial pr description, I surely should have done that. It is still open to discussion.

MS no longer exposes "no associated draft" as a text node by default. Instead, the text node is empty. Furthermore, the "no associated draft" patch meant that it would be never announced when a message has an associated draft. Therefore, I propose people who have the "no associated draft" issue updating their outlook to fix this, rather than keeping the old patch. CC @bramd

Reply:
I concur.